src/libraries/System.IO.Ports/tests/SerialPort/ReadTo.cs

```diff
@@ -861,10 +864,14 @@ private string GenRandomNewLine(bool validAscii)
 
         private int GetUTF7EncodingBytes(char[] chars, int index, int count)
         {
-            byte[] bytes = Encoding.UTF7.GetBytes(chars, index, count);
+#pragma warning disable BCL0001 // Encoding.UTF7 property is obsolete
```

Comment:
>BCL0001 [](start = 24, length = 7)

I am seeing you are disabling the warning in 2 places in this project. can we cache Encoding.UTF7 in one place and disable the warning there and then reference the cached values else where?